src/ssh/cockpitsshknownhosts.c

```diff
@@ -17,16 +17,147 @@
  * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "cockpitsshknownhosts.h"
+
+#include <glib/gstdio.h>
+#include <glib.h>
+#include <glib-object.h>
+
 #include <ctype.h>
 #include <string.h>
 #include <errno.h>
 #include <sys/types.h>
 #include <stdio.h>
 
-#include <glib/gstdio.h>
-#include "cockpitknownhosts.h"
+static gchar *knownhosts_file = NULL;
+
+void shim_set_knownhosts_file (const gchar *file)
+{
+  knownhosts_file = g_strdup (file);
```

Comment:
This should `g_free (knownhosts_file)` first, in case this is being called multiple times.